packages/syft/src/syft/core/tensor/lazy_repeat_array.py

```diff
@@ -83,32 +102,59 @@ def __pow__(self, exponent: int) -> lazyrepeatarray:
             return self * self
         raise Exception("not sure how to do this yet")
 
-    def simple_assets_for_serde(self) -> Tuple[np.ndarray, Tuple[int, ...]]:
-        return (self.data, self.shape)
-
-    @staticmethod
-    def deserialize_from_simple_assets(
-        assets: Tuple[np.ndarray, Tuple[int, ...]]
-    ) -> lazyrepeatarray:
-        return lazyrepeatarray(data=assets[0], shape=assets[1])
-
     def copy(self, order: Optional[str] = "K") -> lazyrepeatarray:
-        return lazyrepeatarray(data=np.copy(self.data, order=order), shape=self.shape)
+        return self.__class__(data=np.copy(self.data, order=order), shape=self.shape)
 
     @property
     def size(self) -> int:
         return np.prod(self.shape)
 
     def sum(self, axis: Optional[int] = None) -> np.ndarray:
         if axis is None:
-            if self.data.size == 1:
-                return np.array(self.data * self.size).flatten()
-            else:
-                raise Exception("not sure how to do this yet")
+            # TODO: make fast
+            return self.to_numpy().sum()
         else:
             raise Exception("not sure how to do this yet")
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any) -> lazyrepeatarray:  # type: ignore
         if isinstance(other, lazyrepeatarray):
-            return self.data == other.data and self.shape == other.shape
+            if self.shape == other.shape:
+                return lazyrepeatarray(data=self.data == other.data, shape=self.shape)
```

Comment:
Q: Should we not return a boolean , here why are we wrapping in a lazy repeat array.